plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -558,25 +558,48 @@ lookupVarName varName@(NamedDeBruijn _ varIx) varEnv =
             var = Var () varName
         Just val -> pure val
 
+{- Note [Laziness of evalBuiltinApp]
```

Comment:
Can you mention the `Lazy` PR here and explain why that approach is less performant (if you know the answer)